yt_idefix/data_structures.py

```diff
@@ -724,3 +797,241 @@ def _validate_units_override_keys(cls, units_override):
+            txt = gridtxt.readlines()
+            self.domain_left_edge = np.array(
+                [
+                    float(
+                        txt[i]
+                        .replace(",", "")
+                        .replace("[", "")
+                        .replace("]", "")
+                        .split()[2]
+                    )
+                    for i in [6, 7, 8]
+                ]
+            )
+            self.domain_right_edge = np.array(
+                [
+                    float(
+                        txt[i]
+                        .replace(",", "")
+                        .replace("[", "")
+                        .replace("]", "")
+                        .split()[3]
+                    )
+                    for i in [6, 7, 8]
+                ]
+            )
+            self.dimensionality = int(txt[4][-2])
+            self.domain_dimensions = np.array(
+                [
+                    int(
+                        txt[i]
+                        .replace(",", "")
+                        .replace("[", "")
+                        .replace("]", "")
+                        .split()[4]
+                    )
+                    for i in [6, 7, 8]
+                ][: self.dimensionality]
+            )
+
+            def parse_geometry(geom: str):
+                import yt
+
+                if yt.version_info[:2] > (4, 1):
+                    try:
+                        from yt.geometry.api import (
+                            Geometry,  # type: ignore [attr-defined]
+                        )
+
+                        return Geometry(geom)
+                    except ImportError:
+                        pass
+
+                return geom
+
+            geom_str = (txt[5].split()[-1]).lower()
+            self.geometry = parse_geometry(geom_str)
+
+        with open(out_file) as outttxt:
+            txt = outttxt.readlines()
+            entry = int(
+                os.path.basename(self.parameter_filename)
+                .replace(".flt.h5", "")
+                .replace("data.", "")
+                .replace(".dbl.h5", "")
+            )
+            self.current_time = float(txt[entry].split()[1])
+            self.ntracers = 0
+            search = "tr1"
+            while search in txt[entry].split():
+                self.ntracers += 1
+                search = "tr%d" % (self.ntracers + 1)
+
+        self.fluid_types += (self._dataset_type,)
+        self.gamma = 5.0 / 3.0
+        self.mu = 0.61
+        self.storage_filename = self.parameter_filename
+        self.refine_by = 1  # no mesh refinement
+        self._periodicity = (True, True, True)
+        # PlutoStatic cannot yet be run as a cosmological simulation
+        self.cosmological_simulation = 0
+        self.current_redshift = 0.0
+        self.omega_lambda = 0.0
+        self.omega_matter = 0.0
+        self.hubble_constant = 0.0
+
+    def _set_code_unit_attributes(self):
```

Comment:
Is there a reason to override PlutoVtkDataset's implementation of this method ?
They seem to accomplish the same thing, so I'd prefer to keep them identical unless they're an actual difference I am missing

Reply:
I trust @xshaokun could answer that.

Reply:
> I'm not sure if the logic of partial code units in definitions.h is already implemented.

This method in PlutoVtkDataset is able to handle the partial definitions of these three units. Namely, if any of them is not defined in definitions.h, the default value will be assigned as Pluto did. 

However I just realized the default values are wrong. Thanks for your remind! I will fix it in my next PR soon.

Reply:
Feel free to open a separate PR for that.

Reply:
Yes this was the reason I had a separate code to parse the ```definitions.h```. Maybe my part of the code can be put in one of the parent classes to avoid code repetitions. @xshaokun and @dutta-alankar Please have a look around this region of code in ```PlutoXdmfDataset._set_code_unit_attributes(self)```.

Reply:
As I see, your method is doing almost the same things as `PlutoVtkDataset._set_code_unit_attributs(self)`, except that unit definitions in `definitions.h` was parsed in `PlutoVtkDataset._parse_definitions_header(self)`

Reply:
I suggest we move discussion to #204 and then come back here and simplify this one :)

Reply:
Addressed!